spec/models/location_spec.rb

```diff
@@ -119,6 +119,34 @@
       Timecop.freeze(2022, 6, 7)
     end
 
+    context "when location is missing some mandatory information" do
+      it "returns incomplete when the postcode is missing" do
+        location.postcode = nil
+        expect(location.status).to eq(:incomplete)
+        location.postcode = ' '
```

Comment:
```suggestion
        location.postcode = ""
```

Reply:
@SamSeed-Softwire You are testing a case that isn't possible. Try saving a postcode with a whitespace postcode:

```ruby
irb(main):006:0> location = Location.new(postcode: "   "); location.valid?
=> false
irb(main):007:0> location.errors[:postcode]
=> ["Enter a postcode in the correct format, for example AA1 1AA"]
```

Reply:
@jamesrose thanks. I was trying to write a test under the assumption that the postcode validation might not be working correctly. But I suppose there's already an explicit test to make sure it is:

```
    it "does add an error when the postcode is invalid" do
      location.postcode = "invalid"
      expect { location.save! }
        .to raise_error(ActiveRecord::RecordInvalid, "Validation failed: Postcode #{I18n.t('validations.postcode')}")
    end
```